2996_reflection/reflection.md

```diff
@@ -402,6 +464,16 @@ Note that this last version has lower complexity: While the versions using an ex
 
 [On Compiler Explorer](https://godbolt.org/z/Y5va8MqzG).
 
+
+Many many variations of these functions are possible and beneficial depending on the needs of the client code.
+For example:
+
+  - the "<unnamed>" case could instead output a valid cast expression like "E(5)"
+  - a more sophisticated lookup algorithm could be selected at compile time depending on the length of `enumerators_of(^E)`
```

Comment:
I think it's worth actually doing this in the example - will do so later after I merge this.